crates/types/src/net/mod.rs

```diff
@@ -96,17 +96,28 @@ impl AdvertisedAddress {
     pub fn derive_bind_address(&self) -> BindAddress {
         match self {
             AdvertisedAddress::Http(uri) => {
-                let port = uri
-                    .authority()
-                    .and_then(|auth| auth.port_u16())
-                    .unwrap_or(80); // HTTP default port is 80 if unspecified
-
                 let ip = if uri.host().unwrap_or("").contains(':') {
                     IpAddr::V6(Ipv6Addr::UNSPECIFIED)
                 } else {
                     IpAddr::V4(Ipv4Addr::UNSPECIFIED)
                 };
 
+                let default_port = 5122;
+                let port = uri
+                    .authority()
+                    .and_then(|auth| auth.port_u16())
+                    .unwrap_or_else(|| {
+                        // if advertised-address is in the form "http://node" we will bind to :5122
+                        if uri.scheme().is_some() {
+                            warn!(
+                                "The node advertised address \"{self}\" does not specify a port. \
+                                Restate will bind to {ip}:{default_port} - please ensure that peer \
+                                nodes can communicate with this node using its advertised address"
+                            );
+                        };
+                        default_port
+                    });
```

Comment:
If we default to 5122 then, we should also make sure that the advertised address that we write into the `NodesConfiguration` gets this port. Otherwise, clients will try to connect to `http://node` but the server is bound to `0.0.0.0:5122`.

Alternatively, we could print a warning when defaulting to port 80.

Reply:
Yup! This is only half the story, hence the draft status of the PR - I think we want to also make a noise if advertised-address doesn't have an explicit port set, and have a default there. The change as it stands is not sufficient to resolve the gotcha.